enterprise/kernel/src/main/java/org/neo4j/kernel/impl/pagecache/Profile.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2002-2018 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.pagecache;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Objects;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+import java.util.zip.GZIPInputStream;
+import java.util.zip.GZIPOutputStream;
+
+import org.neo4j.io.IOUtils;
+import org.neo4j.io.fs.FileSystemAbstraction;
+import org.neo4j.io.pagecache.PagedFile;
+
+import static org.neo4j.kernel.impl.pagecache.PageCacheWarmer.SUFFIX_CACHEPROF;
+
+final class Profile implements Comparable<Profile>
+{
+    private final File profileFile;
+    private final File pagedFile;
+    private final long profileCount;
+
+    private Profile( File profileFile, File pagedFile, long profileCount )
+    {
+        Objects.requireNonNull( profileFile );
+        Objects.requireNonNull( pagedFile );
+        this.profileFile = profileFile;
+        this.pagedFile = pagedFile;
+        this.profileCount = profileCount;
+    }
+
+    @Override
+    public int compareTo( Profile that )
+    {
+        int compare = profileFile.compareTo( that.profileFile );
```

Comment:
You probably don't want to sort them lexicographically since that would result in
```
profile1
profile10
profile2
profile3
etc..
```
If this is not the case, we should at least have a test for that.